h/search/query.py

```diff
@@ -165,6 +165,33 @@ def __call__(self, search, _):
         return search.filter("terms", group=groups)
 
 
+class UriWildcardFilter(object):
+
+    """
+    A filter that selects only annotations where the 'site' parameter matches as *site*.
+    """
+
+    def __init__(self, request):
+        """Searches for wildcard uri's.
+
+        :param request: the pyramid.request object
+
+        """
+        self.request = request
+
+    def __call__(self, search, params):
+        query_uris = params.getall('wildcard_uri')
+
+        uris = set()
+        for query_uri in query_uris:
+            expanded = storage.expand_uri(self.request.db, query_uri)
```

Comment:
The `query_uri` presumably won't match the URLs stored in the index because it includes wildcard `*` markers?